Source/Scene/Model.js

```diff
@@ -4094,20 +4018,32 @@ define([
                 }
 
                 this._loadResources = new LoadResources();
-                parse(this);
             }
+            parse(this);
         }
 
         var loadResources = this._loadResources;
         var incrementallyLoadTextures = this._incrementallyLoadTextures;
         var justLoaded = false;
 
         if (this._state === ModelState.LOADING) {
-            // Create WebGL resources as buffers/shaders/textures are downloaded
-            createResources(this, frameState);
-
             // Transition from LOADING -> LOADED once resources are downloaded and created.
             // Textures may continue to stream in while in the LOADED state.
+            if (loadResources.pendingBufferLoads === 0 && loadResources.pendingShaderLoads === 0) {
+                if (!this._updatedGltfVersion) {
+                    updateVersion(this.gltf);
```

Comment:
> Should this updateVersion be one of the first things that happens?

The reason it is down here is because buffers need to be loaded before we use `updateVersion`. However, I do agree that `addDefaults` and `processModelMaterialsCommon` should execute after. I will need to break up `parse` a bit more here to make that work, but I think that's a reasonable approach.

> This will, of course, require more gltf-pipeline changes

Yup, I'm going to work on those today after I migrate the changes back from here.

> why is addPipelineExtras required? Because the other gltf-pipeline stages require it, right? Why are we also using extras for loading resources here?

addPipelineExtras makes sure that there is somewhere to put the reference to things like shader sources and buffer data in the way that gltf-pipeline code expects. When the model loader code pulls a resource it also adds a reference to the data here. This was the cleanest approach I could find to be able to use pipeline code without modifying the resource loader.

> Are the extras removed later?

We can, we do have a stage for that. The only compelling reason I can think of to do that would be if you were freeing the resource data, the gltf object still has references to the buffers, shaders etc, so the data wouldn't be freed. I don't know that we ever do that though.

> Also, is the _updatedGltfVersion flag needed?

Yes it is. This part of the code may be called multiple times and we only want to run updateVersion once.